clash-lib/src/Clash/Normalize/Strategy.hs

```diff
@@ -137,6 +138,7 @@ constantPropagation =
     specTransformations =
       [ ("typeSpec"    , typeSpec)
       , ("nonRepSpec"  , nonRepSpec)
+      , ("zeroWidthSpec", zeroWidthSpec)
```

Comment:
The CL says:

> the normalizer is then free to further normalize this to

But that only happens accidentally, as a result of e.g. `caseCon` being part of `anf`. Or as a result of the late-stage function hierarchy flattening/inline that happens after regular normalization.

It's worth adding a note wrt the above.

Reply:
I've left a source note in the `Strategy` module for this

Reply:
Thanks, LGTM